paddle/phi/infermeta/spmd_rules/conv2d.cc

```diff
@@ -0,0 +1,203 @@
+    int output_ndim = static_cast<int>(original_output_shape.size());
+    PADDLE_ENFORCE_EQ(
+        output_ndim,
+        4,
+        common::errors::InvalidArgument("The Tensor Output's rank must be 4 in conv2d, for NCoutHW format."
+                                        "But now it's [%d]",
+                                        output_ndim));
+    
+    auto output_dist_attr_src = output.dist_attr();
+    std::vector<int64_t> output_dims_mapping = output_dist_attr_src.dims_mapping();
+
+    // Step1: build Einsum Notation
+    // todo: check output notation, how to deal with the "Input HW, Filter HW and Output HW"...
+    VLOG(4) << "step 1: build Einsum Notation";
+    std::string input_axes = "nchw";
+    std::string filter_axes = "mcij";
+    std::string output_axes = "nmhw";
+
+    // Step2: sharding propagation
+    auto axis_to_dim_map = ShardingMergeForTensors({{output_axes, output_dims_mapping}}, false);
+    TensorDistAttr input_dist_attr_dst = CopyTensorDistAttrForOutput(input.dist_attr());
+    TensorDistAttr filter_dist_attr_dst = CopyTensorDistAttrForOutput(filter.dist_attr());
+    input_dist_attr_dst.set_dims_mapping(GetDimsMappingForAxes(input_axes, axis_to_dim_map, true));
+    filter_dist_attr_dst.set_dims_mapping(GetDimsMappingForAxes(filter_axes, axis_to_dim_map, true));
+
+    VLOG(4) << "Conv2dReverseSPMDRule InferBackward: "
+          << "Einsum notation: [" << input_axes << "," << filter_axes << " --> "
+          << output_axes << "]. " << std::endl;
+    LogInputDistAttr("Output", original_output_shape, output_dist_attr_src, output_dist_attr_src);
+    LogOutputDistAttr("Input" , input_dist_attr_dst);
+    LogOutputDistAttr("Filter", filter_dist_attr_dst);
+    VLOG(4) << std::endl;
+
+    return {{input_dist_attr_dst, filter_dist_attr_dst}, {output_dist_attr_src}};
+}
+
+SpmdInfo Conv2dInferSpmd(const DistMetaTensor& input,
+                            const DistMetaTensor& filter,
+                            const std::vector<int>& strides,
+                            const std::vector<int>& paddings,
+                            const std::string& padding_algorithm,
+                            const std::vector<int>& dilations,
+                            int groups,
+                            const std::string& data_format) {
+    return Conv2dInferSpmdBase(input, filter);
+}
+
+
+SpmdInfo Conv2dInferSpmdReverse(const DistMetaTensor& input,
+                            const DistMetaTensor& filter,
+                            const DistMetaTensor& output,
+                            const std::vector<int>& strides,
+                            const std::vector<int>& paddings,
+                            const std::string& padding_algorithm,
+                            const std::vector<int>& dilations,
+                            int groups,
+                            const std::string& data_format) {
+        return Conv2dInferSpmdReverseBase(input, filter, output);
+    }
+
```

Comment:
shall we support `Conv2dGradInferSpmdBase` and `Conv2dGradInferSpmd`, or auto-parallel in dygraph will not support